Check against multiple versions

2m opened the pull request:
Resolves #16

Builds on top of https://github.com/typesafehub/migration-manager/pull/63

Can be tested by
```
addSbtPlugin("com.typesafe" % "sbt-mima-plugin" % "0.1-695c6b81a6962c726983c667eb3393b4840f8f39")
resolvers += Resolver.url("2m-sbt-plugins", url("https://dl.bintray.com/2m/sbt-plugins/"))(Resolver.ivyStylePatterns)
```

2m commented:
Had to update sbt only up to `0.13.6` instead of `0.13.8`, as otherwise this plugin did not work with akka 2.3 which is being built with `0.13.5` (release-2.3 branch is not to be upgraded to the latest sbt).

```
[error] (akka-zeromq/*:mimaPreviousClassfiles) java.lang.NoSuchMethodError: sbt.ModuleInfo$.apply$default$9()Lscala/collection/Seq;
[error] (akka-kernel/*:mimaPreviousClassfiles) java.lang.NoSuchMethodError: sbt.ModuleInfo$.apply$default$9()Lscala/collection/Seq;
[error] (akka-dataflow/*:mimaPreviousClassfiles) java.lang.NoSuchMethodError: sbt.ModuleInfo$.apply$default$9()Lscala/collection/Seq;
[error] (akka-slf4j/*:mimaPreviousClassfiles) java.lang.NoSuchMethodError: sbt.ModuleInfo$.apply$default$9()Lscala/collection/Seq;
[error] (akka-persistence-experimental/*:mimaPreviousClassfiles) java.lang.NoSuchMethodError: sbt.ModuleInfo$.apply$default$9()Lscala/collection/Seq;
[error] (akka-file-mailbox/*:mimaPreviousClassfiles) java.lang.NoSuchMethodError: sbt.ModuleInfo$.apply$default$9()Lscala/collection/Seq;
[error] (akka-mailboxes-common/*:mimaPreviousClassfiles) java.lang.NoSuchMethodError: sbt.ModuleInfo$.apply$default$9()Lscala/collection/Seq;
[error] (akka-cluster/*:mimaPreviousClassfiles) java.lang.NoSuchMethodError: sbt.ModuleInfo$.apply$default$9()Lscala/collection/Seq;
[error] (akka-transactor/*:mimaPreviousClassfiles) java.lang.NoSuchMethodError: sbt.ModuleInfo$.apply$default$9()Lscala/collection/Seq;
[error] (akka-testkit/*:mimaPreviousClassfiles) java.lang.NoSuchMethodError: sbt.ModuleInfo$.apply$default$9()Lscala/collection/Seq;
[error] (akka-actor/*:mimaPreviousClassfiles) java.lang.NoSuchMethodError: sbt.ModuleInfo$.apply$default$9()Lscala/collection/Seq;
[error] (akka-multi-node-testkit/*:mimaPreviousClassfiles) java.lang.NoSuchMethodError: sbt.ModuleInfo$.apply$default$9()Lscala/collection/Seq;
[error] (akka-agent/*:mimaPreviousClassfiles) java.lang.NoSuchMethodError: sbt.ModuleInfo$.apply$default$9()Lscala/collection/Seq;
[error] (akka-camel/*:mimaPreviousClassfiles) java.lang.NoSuchMethodError: sbt.ModuleInfo$.apply$default$9()Lscala/collection/Seq;
[error] (akka-remote/*:mimaPreviousClassfiles) java.lang.NoSuchMethodError: sbt.ModuleInfo$.apply$default$9()Lscala/collection/Seq;
```

dotta commented:
@jsuereth Considering I know pretty much nothing about the sbt mima plugin, could you have a look at it and give your bless :-)

(sorry Josh for spamming you on the other PR ;-))

2m commented:
I have removed the now one superfluous commit and also changed it so it is not a build breaking change anymore. I deprecated `previousArtifact` and added new `previousArtifacts`.

2m commented on project/Build.scala:
```diff
@@ -27,7 +27,6 @@ object BuildSettings {
   val commonSettings = Defaults.defaultSettings ++ Seq (
       organization := buildOrganization,
       scalaVersion := buildScalaVer,
-      git.baseVersion := "0.1",
```
This seems to be unsused, because `git.useGitDescribe := true` is now set.

ktoso replied:
makes sense

ktoso commented:
Looks good I think :-)

dotta commented:
@ktoso @2m I'm busy with a Scala training today and tomorrow. I'll have a look at it next week. Sorry guys.

ktoso commented:
No problem, it's not super urgent.

Would be nice if one of us (team) could also release this project in the
future :-)
On Jun 18, 2015 6:59 PM, "Mirco Dotta" <notifications@github.com> wrote:

> @ktoso <https://github.com/ktoso> @2m <https://github.com/2m> I'm busy
> with a Scala training today and tomorrow. I'll have a look at it next week.
> Sorry guys.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/typesafehub/migration-manager/pull/64#issuecomment-113220520>
> .
>

dotta commented:
> Would be nice if one of us (team) could also release this project in the
future :-)

I'd be extremely happy with that! Let me know who wants to be blessed with this honour? :-)

ktoso commented:
Can you add @2m and me maybe?

dotta commented:
Done - you are both admin for this project now

ktoso commented:
Thanks a lot!
On Jun 23, 2015 7:31 PM, "Mirco Dotta" <notifications@github.com> wrote:

> Done - you are both admin for this project now
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/typesafehub/migration-manager/pull/64#issuecomment-114581794>
> .
>
